Refactor/issue #17980

noeliaSD opened the pull request:
Closes #17980

### What does the PR do

- It removes direct calls to `utils store` in `MessageView` component and its hierarchy.

### Affected areas

Chat

### Architecture compliance

- [X] I am familiar with the application architecture and agreed good practices.
My PR is consistent with this document: [Status Desktop Architecture Guide](https://github.com/status-im/status-desktop/blob/master/CONTRIBUTING.md)

### Impact on end user

None

### How to test

- Chat module

### Risk 

Tick **one**:
- [ ] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [X] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e5abc889 | [#2](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17981/2/) | 2025-05-21 14:47:08 | ~13 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-17981/2//consoleText) |

caybro commented on ui/app/AppLayouts/Chat/ChatLayout.qml:
```diff
@@ -65,6 +64,10 @@ StackLayout {
     property bool neverAskAboutUnfurlingAgain
 
+    // Utils store data
+    property var cbGetCompressedPk: function () {}
```
```suggestion
    property var cbGetCompressedPk: function (publicKey) { console.error("Implement me"); return ""}
```

To suggest what's the signature and return type; and make it "required"

caybro replied:
Same below

caybro commented on ui/app/AppLayouts/Chat/views/ChatColumnView.qml:
```diff
@@ -63,6 +62,10 @@ Item {
     property bool neverAskAboutUnfurlingAgain
 
+    // Utils store data
+    property var cbGetCompressedPk: function () {}
```
Same here and anywhere else below

caybro commented on ui/imports/shared/stores/UtilsStore.qml:
```diff
@@ -30,6 +30,7 @@ QtObject {
     }
 
     function getEmojiHash(publicKey) {
+        console.log("\n\n++++++++++++++++++getEmojiHash")
```
leftover?

caybro commented on ui/imports/shared/stores/UtilsStore.qml:
```diff
@@ -41,6 +42,7 @@ QtObject {
     }
 
     function getCompressedPk(publicKey) {
+        console.log("\n\n++++++++++++++++++getCompressedPk")
```
debug leftover?

caybro reviewed (commented):
Looks good in general, just some minor remarks

micieslak commented on ui/imports/shared/views/chat/LinksMessageView.qml:
```diff
@@ -113,8 +114,7 @@ Flow {
 
-            utilsStore: root.utilsStore
-
+            cbGetEmojiHash: root.cbGetEmojiHash
```
it seems that every instance created by the repeater takes the same `cbGetEmojiHash` and calls it for the same pub key. Hopefully,we could take the emoji hash here directly.

micieslak replied:
Probably `LiksMessagePreview` can just take the emoji hash directly, not via a function.

micieslak replied:
Ok, I missed the fact that it's dependent on the links model content. Then the cleanest way is probably to provide that hash via model in `link_preview_model.nim`.

micieslak reviewed (commented):
LGTM, just some doubts/questions to consider

noeliaSD commented on src/app/modules/shared_models/message_model.nim:
```diff
@@ -361,6 +364,12 @@ QtObject:
+        result = newQVariant("")
+      if not (singletonInstance.utils.isChatKey(item.senderId)):
+          result = newQVariant(item.senderId)
+      result = newQVariant(singletonInstance.utils.getCompressedPk(item.senderId))
```
Totally agree!! Thanks for the suggestion! Pushed changes addressing it! Please, take a look

noeliaSD commented:
@micieslak please take a second look. I added 2 commits on top that add the roles to the corresponding models and then, removed the function calls all over the chain! Much cleaner and makes more sense! Thanks for the suggestion!

@jrainville also, please take a second look!

jrainville reviewed (approved):
Nice!

micieslak reviewed (approved):
Nice work! No more comments from my side :)
